plutus-core-interpreter/test/DynamicBuiltins/MakeRead.hs

```diff
@@ -22,12 +24,21 @@ import qualified Hedgehog.Gen                               as Gen
 import qualified Hedgehog.Range                             as Range
 import           Test.Tasty
 import           Test.Tasty.Hedgehog
+import           Test.Tasty.HUnit
+
+readMakeHetero
+    :: (KnownDynamicBuiltinType a, KnownDynamicBuiltinType b)
+    => a -> Maybe (Either CekMachineException b)
+readMakeHetero = makeDynamicBuiltin >=> sequence . readDynamicBuiltinCek
+
+readMake
+    :: KnownDynamicBuiltinType a => a -> Maybe (Either CekMachineException a)
+readMake = readMakeHetero
```

Comment:
?

Reply:
"?"? :)

Below in `test_ignoreEvaluationFailure` we read back a value of a different type than the type of a "made" value. Thus `readMakeHetero` allows to do that, while `readMake` requires the original and the final type to be equal, because otherwise we'd get type ambiguity.